tests/e2e/pageobjects/ide/Ide.ts

```diff
@@ -41,6 +41,57 @@ export class Ide {
         await this.driverHelper.waitAndSwitchToFrame(By.css(Ide.IDE_IFRAME_CSS), timeout);
     }
 
+    async waitDialogShell(timeout: number, openLink: boolean, closeDialog: boolean) {
+        const dialogShellLocator: By = By.id('theia-dialog-shell');
```

Comment:
I think that this locator is redundant. So it can be included to the `dialogBlockLocator`: `//div[@id="theia-dialog-shell"]/div[@class="dialogBlock"]`.
So instead of having this:
```
        const dialogShellLocator: By = By.id('theia-dialog-shell');
        const dialogBlockLocator: By = By.xpath('.//div[@class="dialogBlock"]');
         [other locators]
        Logger.debug(`Ide.waitDialogShell timeout:"${timeout}"`);

        Logger.debug(`Ide.waitDialogShell waiting for dialog shell to become visible.`);
        await this.driverHelper.waitVisibility(dialogShellLocator, timeout);

        Logger.debug(`Ide.waitDialogShell get dialog shell.`);
        const dialogShell: WebElementPromise = (await this.driverHelper.getDriver()).findElement(dialogShellLocator);
        Logger.debug(`Ide.waitDialogShell finding dialogBlock child element.`);
        const dialogBlock: WebElementPromise = (await dialogShell).findElement(dialogBlockLocator);
```

we can have just this:
```
const dialogBlockLocator: By = By.xpath('//div[@id="theia-dialog-shell"]/div[@class="dialogBlock"]');
        const dialogBlockTitleLocator: By = By.xpath('.//div[@class="dialogTitle"]/div');
        [other locators]

        const dialogBlock = this.driverHelper.waitVisibility(dialogBlockLocator, timeout);
```

Reply:
Simplified the locators a lot, removed unnecessary element detection.